src/Compilers/Core/CodeAnalysisTest/DefaultAnalyzerAssemblyLoaderTests.cs

```diff
@@ -30,63 +30,131 @@
 
 namespace Microsoft.CodeAnalysis.UnitTests
 {
-    [CollectionDefinition(Name)]
-    public class AssemblyLoadTestFixtureCollection : ICollectionFixture<AssemblyLoadTestFixture>
-    {
-        public const string Name = nameof(AssemblyLoadTestFixtureCollection);
-        private AssemblyLoadTestFixtureCollection() { }
-    }
+#if NETCOREAPP
 
-    public sealed class DefaultAnalyzerAssemblyLoaderTests : TestBase
+    public sealed class InvokeUtil
     {
-#if NETCOREAPP
-        private sealed class InvokeUtil
+        public void Exec(Action<string> testOutputHelper, AssemblyLoadContext alc, bool shadowLoad, string typeName, string methodName)
         {
-            public void Exec(string typeName, string methodName) => InvokeTestCode(typeName, methodName);
+            // Ensure that the test did not load any of the test fixture assemblies into 
+            // the default load context. That should never happen. Assemblies should either 
+            // load into the compiler or directory load context.
+            //
+            // Not only is this bad behavior it also pollutes future test results.
+            var count = AssemblyLoadContext.Default.Assemblies.Count();
+            using var fixture = new AssemblyLoadTestFixture();
+            using var tempRoot = new TempRoot();
+            var loader = shadowLoad
+                ? new ShadowCopyAnalyzerAssemblyLoader(alc, tempRoot.CreateDirectory().Path)
+                : new DefaultAnalyzerAssemblyLoader(alc);
+            try
+            {
+                DefaultAnalyzerAssemblyLoaderTests.InvokeTestCode(loader, fixture, typeName, methodName);
+            }
+            finally
+            {
+                testOutputHelper($"Test fixture root: {fixture.TempDirectory.Path}");
+
+                foreach (var context in loader.GetDirectoryLoadContextsSnapshot())
+                {
+                    testOutputHelper($"Directory context: {context.Directory}");
+                    foreach (var assembly in context.Assemblies)
+                    {
+                        testOutputHelper($"\t{assembly.FullName}");
+                    }
+                }
+
+                if (loader is ShadowCopyAnalyzerAssemblyLoader shadowLoader)
+                {
+                    testOutputHelper($"Shadow loader: {shadowLoader.BaseDirectory}");
+                }
+
+                testOutputHelper($"Loader path maps");
+                foreach (var pair in loader.GetPathMapSnapshot())
+                {
```

Comment:
It may aid readability to quote the values, e.g. `$"\t""{pair.Key}"" -> ""{pair.Value}"""`